doc/source/building/index.rst

```diff
@@ -363,6 +370,14 @@ like build the html documentation or running benchmarks. The ``spin``
 interface is self-documenting, so please see ``spin --help`` and
 ``spin <subcommand> --help`` for detailed guidance.
 
+.. warning::
+
+    When building in Windows, `a bug in meson <https://github.com/mesonbuild/meson/issues/10022>`_ may cause the wrong build
+    tools to be selected, which manifests as a LNK1107 fatal error being
+    emitted by link.exe. To workaround, set environment variables CC_LD
+    and CXX_LD to lld-link.
```

Comment:
This isn't correct for MSVC but specific to `clang-cl` usage I believe. So I'm not sure about including this one.

Reply:
I've setup the numpy-dev environment on two different Windows 10 machines with different versions of Visual Studio 2022, and in both cases the spin build process detects cl.exe and link.exe as the compiler and linker, but then still uses llvm-ar to create .a files that link.exe cannot read. Per the meson issue link above, this incorrect tool flow selection problem seems to be a meson bug. The workaround I could make work was setting environment variables such that lld-link, which does read .a files, is used instead of link.exe. I don't doubt that this isn't the ideal solution, but I don't understand the meson build process well enough to debug what's going wrong such that llvm-ar is getting involved in a MSVC build flow. Any pointers on where to look to fix this flow more correctly would be much appreciated. Right now, it appears any fresh Windows build setup, at least with Visual Studio 2022 (have not checked other versions), is broken, and the lld-link workaround is the only path I've found to fix it.

Reply:
Hmm, I'm not 100% sure. Could be that Conda is setting the `AR` env var perhaps (not sure). 

> Right now, it appears any fresh Windows build setup, at least with Visual Studio 2022 (have not checked other versions),

Pretty sure it's working for me with Pixi, but I haven't tried with regular Conda env activation in quite a while (should be the same though).

I'm okay with keeping the note then.

Reply:
Ok, Eureka! Looking through the numpy-dev conda activation scripts in numpy-dev/etc/conda/activate.d/, I see flang_win-64_activate.bat sets environment variable AR=llvm-ar.exe. Unsetting this variable fixes the build! The same script also sets LDFLAGS to add a clang specific option which triggers unrecognized options warnings during the build and sets LD=lld-link.exe, which doesn't seem to matter, but shouldn't be there.

This script is getting added via the "compilers" package entry in environment.yml. This package is required to get the vcvars auto-setup to work (and maybe for other reasons), so it shouldn't be removed. As far as I can tell, there isn't a way to specify package options to get a version of compilers without flang. I don't know if there's some other good way to get conda to keep everything required installed but just not run the flang activation scripts?

Assuming no smoother solutions are on offer, I propose that I delete this proposed section and instead add a comment in both the "Building from source to use NumPy" and "Building from source for NumPy development" sections recommending unsetting AR, LDLAGS, and LD before running spin build on Windows.

Reply:
> This script is getting added via the "compilers" package entry in environment.yml.

This didn't used to work at all on Windows until very recently, so one had to comment out `compilers` to get the environment to install. That's perhaps why you're the first one being bitten by it.

> and instead add a comment in both the "Building from source to use NumPy" and "Building from source for NumPy development" sections recommending unsetting AR, LDLAGS, and LD before running spin build on Windows

That does sound nicer indeed.